interview/algorithm/naive_bayes.py

```diff
@@ -0,0 +1,108 @@
+from collections import Counter
+
+import numpy as np
+
+
+class NaiveBayes:
+    """Naive Bayes methods are a set of supervised learning algorithms based on
+    applying Bayes’ theorem with the “naive” assumption of conditional independence
+    between every pair of features given the value of the class variable.
+    """
+
+    def __init__(self):
+        self.x_train = None
+        self.y_train = None
+        self.classes = None
+
+    def fit(self, x_train, y_train):
+        """Store training dataset.
+
+        :param list x_train: training dataset. Each training example has several
+            features.
+        :param list y_train: outcomes of training dataset.
+        """
+        self.x_train = x_train
+        self.y_train = y_train
+        self.classes = list(set(y_train))
+
+    def arrange(self):
+        """Group training examples by class
+
+        :return: (*dict*) -- dictionary giving training examples for each class.
+        """
+        return {c: self.x_train[np.where(self.y_train == c)] for c in self.classes}
```

Comment:
Another way to do this is using `itertools.groupby`

Reply:
Yes but I am not sure it would be shorter.